assets/src/edit-story/app/story/effects/useHistoryEntry.js

```diff
@@ -29,15 +29,25 @@ function useHistoryEntry({ story, current, pages, selection, capabilities }) {
   const {
     actions: { appendToHistory },
   } = useHistory();
+
+  const currentPageIndexRef = useRef();
+  const selectedElementIdsRef = useRef();
+  const pagesRef = useRef();
+  useEffect(() => {
+    currentPageIndexRef.current = current;
+    selectedElementIdsRef.current = selection;
+    pagesRef.current = pages;
+  }, [current, selection, pages]);
+
   useEffect(() => {
     appendToHistory({
       story,
-      current,
+      current: currentPageIndexRef.current,
+      selection: selectedElementIdsRef.current,
       pages,
-      selection,
       capabilities,
     });
-  }, [appendToHistory, story, current, pages, selection, capabilities]);
+  }, [capabilities, story, pages, appendToHistory]);
```

Comment:
This should work in theory, however, `pages` seems to be changing with selection as well, causing the `useEffect` 
 (and `appendToHistory`) to trigger.

Reply:
@dvoytenko FI: It looks like your proposed PR of #621 fixes the issue of `pages` incorrectly being changes/re-initialized and fixes the `pages` issue which blocks this PR.

It probably makes sense to wait until #621 gets merged to merge this one.

Reply:
@miina could you pls clarify whey the pages are incorrectly changed now? Otherwise, certainly ok to block until #621 is merged. I'll try to get it in sooner.

Reply:
> @miina could you pls clarify whey the pages are incorrectly changed now?

@dvoytenko Hmm, actually, it looks like there is no issue with `pages` anymore when testing.
Previously, when testing, creating a new history was triggered even if nothing was actually changed on a Page. Testing showed that `pages` was the variable triggering it. I did not investigate further since it seemed to work OK on #621, however, it seems to work as expected currently as well.

Will mark this ready for review then.

Reply:
Ok. This all looks pretty good then. But why are the capabilities here? Is there an issue already open to move them elsewhere?